vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java

```diff
@@ -759,50 +759,59 @@ public List<String> getAvailableSides() {
   protected String promptForSide() {
     final ArrayList<String> availableSides = new ArrayList<>(sides);
     final ArrayList<String> alreadyTaken = new ArrayList<>();
+    String newSide = "";
 
-    for (final PlayerInfo p : players) {
-      alreadyTaken.add(p.side);
-    }
+    while (newSide != null) { // Loops until a valid side is found or op is canceled (repeats side check to minimuse race condition window)
 
-    availableSides.removeAll(alreadyTaken);
+      for (final PlayerInfo p : players) {
+        alreadyTaken.add(p.side);
+      }
 
-    // If a "real" player side is available, we want to offer "the next one" as the default, rather than observer.
-    // Thus hotseat players can easily cycle through the player positions as they will appear successively as the default.
-    // Common names for Solitaire players (Solitaire, Solo, Referee) do not count as "real" player sides, and will be skipped.
-    // If we have no "next" side available to offer, we stay with the observer side as our default offering.
-    boolean found = false;       // If we find a usable side
-    final String mySide = getMySide(); // Get our own side, so we can find the "next" one
-    final int myidx = (mySide != null) ? sides.indexOf(mySide) : -1; // See if we have a current non-observe side.
-    int i = (myidx >= 0) ? ((myidx + 1) % sides.size()) : 0;   // If we do, start looking in the "next" slot, otherwise start at beginning.
-    for (int tries = 0; i != myidx && tries < sides.size(); i = (i + 1) % sides.size(), tries++) { // Wrap-around search of sides
-      final String s = sides.get(i);
-      if (!alreadyTaken.contains(s) && !isSoloSide(s)) {
-        found = true; // Found an available slot that's not our current one and not a "solo" slot.
+      if (!newSide.isEmpty() && !alreadyTaken.contains(newSide)) {
+        // found side is still available
+        // sides must always be stored internally in English.
```

Comment:
Could you just describe (maybe include in comments in the text) what you are doing here? I think what is happening is that if you choose a side, you then recheck to see if that side has been taken by someone else while you where sitting looking at the change side dialog. 

Have you confirmed that messages from other players are actually being processed (and not just queued up) while the change sides dialog is visible?

Reply:
> Have you confirmed that messages from other players are actually being processed (and not just queued up) while the change sides dialog is visible?

If I understand the question right, then messages (changes?) from other players are processed. So for example;

"Fred" and "Jane" and two players in "<observer>" roles.
Fred attempts to switch side and is presented with the dialogue, offering "Player 1" and "Player 2".
Before Fred makes a choice, Jane also attempts to switch and is offered the same two available sides.
Fred selects "Player 2". Jane at this point still sees both sides available. (Nothing I know of can change that).

The check is done after the player selects a side. So if Jane also selects "Player 2", the re-iterated check detects this (test confirms), and the prompt is re-entered, this time without offering "Player 2".

If I misunderstood your question, please explain what you would like me to check.

Reply:
While Fred is sitting in the Select side dialog and Jane selects Player 1 out from under him and sends a message to Fred saying 'I have selected player 1'. Are you sure that Fred's Vassal client actually processes that message while he is sitting there looking at the select side dialog? When he selects Player 1 also, are you sure that his client will already have been updated with Jane's message? Or will Jane's message only be processed AFTER Fred sends his 'I have select player 1 message'. It's a question of multi-threading and whether anything actually happens in Fred's client while the dialog is open.  

You may have tested this already and know it works?

Reply:
Don't worry, I tried it out. Works well. You see the 'Changed sides' message in the background and the get the 'Unavailable' option when I select the missing side.

Reply:
Jane's message appears on Fred's chat log whilst Fred is still in the dialogue.

In this example, MTB is the other player who switched to Player 2 whilst the current player was in the dialogue. Then current player tries switching to Player 2 and is denied by the new check.

<img width="669" alt="image" src="https://github.com/vassalengine/vassal/assets/58135975/9d25af12-0525-4ece-a92f-6f773ea427c4">

Reply:
Yes, I tried it myself. Vastly improved. Should narrow the race window to the current latency between the two clients in question. The best we can do with out adding a more complex 2 level procedure.